src/redux/actions/samples/createSample.js

```diff
@@ -43,38 +42,59 @@ const createSample = (
       .reduce((acc, curr) => ({ ...acc, [curr]: DEFAULT_NA }), {}) || {},
   };
 
-  const newProject = {
-    uuid: projectUuid,
-    ...project || {},
-    samples: project?.samples.includes(newSampleUuid) ? project.samples
-      : [...project?.samples || [], newSampleUuid],
-  };
+  dispatch({
+    type: SAMPLES_SAVING,
+    payload: {
+      message: endUserMessages.SAVING_SAMPLE,
+    },
+  });
+
+  const url = `/v1/projects/${projectUuid}/${experimentId}/samples`;
 
   try {
-    dispatch(saveSamples(projectUuid, newSample));
-    dispatch(saveProject(projectUuid, newProject));
+    const response = await fetchAPI(
+      url,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newSample),
+      },
+    );
+
+    const json = await response.json();
+
+    throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
 
     dispatch({
       type: SAMPLES_CREATE,
       payload: { sample: newSample },
     });
 
-    dispatch({
-      type: PROJECTS_UPDATE,
-      payload: {
-        projectUuid,
-        project: newProject,
-      },
-    });
-
     dispatch(
       updateExperiment(
         experimentId,
         { sampleIds: [...experiment.sampleIds, newSampleUuid] },
       ),
     );
   } catch (e) {
-    pushNotificationMessage('error', endUserMessages.ERROR_SAVING);
+    let { message } = e;
+    if (!isServerError(e)) {
+      console.error(`fetch ${url} error ${message}`);
+      message = endUserMessages.ERROR_SAVING;
+    }
+
+    dispatch({
+      type: SAMPLES_ERROR,
+      payload: {
+        error: message,
+      },
+    });
+
+    pushNotificationMessage('error', message);
+
+    return Promise.reject(message);
```

Comment:
Instead of returning `Promise.reject(message)`, would it convey the meaning better if we use `throw new Error(message``?